schema/v1.1.0/dataset_config_models_extended.py

```diff
@@ -0,0 +1,964 @@
+from ingestion_tools.scripts.standardize_dirs import IMPORTER_DEP_TREE, flatten_dependency_tree  # noqa: E402
+
+logger = logging.getLogger("dataset-validate")
+
+CELLULAR_COMPONENT_GO_ID = "GO:0005575"
+STRING_FORMATTED_STRING_REGEX = r"^[ ]*\{[a-zA-Z0-9_-]+\}[ ]*$"
+VALID_IMAGE_FORMATS = ("image/png", "image/jpeg", "image/jpg", "image/gif")
+# Note that model namees should all be uppercase or pascal case
+CAMERA_MANUFACTURER_TO_MODEL = {
+    ("FEI", "TFS"): ["FALCON IV", "Falcon4i"],
+    ("Gatan", "GATAN"): ["K2", "K2 SUMMIT", "K3", "K3 BIOQUANTUM", "UltraCam", "UltraScan"],
+}
+FLATTENED_DEP_TREE = {
+    key.type_key: {value.type_key for value in list(value_set)}
+    for key, value_set in flatten_dependency_tree(IMPORTER_DEP_TREE).items()
+}
+
+# Flag to determine if network validation should be run (set by provided Container arg)
+running_network_validation = False
+
+
+# ==============================================================================
+# Author Validation
+# ==============================================================================
+def validate_authors_status(authors: List[Author]) -> List[ValueError]:
+    errors = []
+    primary_author_status_count = 0
+    corresponding_author_status_count = 0
+    for author in authors:
+        if author.primary_author_status:
+            primary_author_status_count += 1
+        if author.corresponding_author_status:
+            corresponding_author_status_count += 1
+
+    if primary_author_status_count == 0:
+        errors.append(ValueError("Annotation must have at least one primary author"))
+    if corresponding_author_status_count == 0:
+        errors.append(ValueError("Annotation must have at least one corresponding author"))
+
+    return errors
+
+
+@alru_cache
+async def lookup_orcid(orcid_id: str) -> Tuple[str, bool]:
+    """
+    Returns a tuple of the ORCID ID and whether or not it is valid
+    """
+    url = f"https://pub.orcid.org/v3.0/{orcid_id}"
+    async with aiohttp.ClientSession() as session, session.head(url) as response:
+        return orcid_id, response.status == 200
+
+
+async def validate_orcids(orcid_list: List[str]) -> List[str]:
+    """
+    Returns a list of invalid ORCIDs, from the provided list
+    """
+    invalid_orcids: List[str] = []
+
+    tasks = [lookup_orcid(orcid) for orcid in orcid_list]
+    results = await asyncio.gather(*tasks)
+    invalid_orcids += [orcid for orcid, valid in results if not valid]
+    return invalid_orcids
+
+
+def validate_authors(authors: List[Author]) -> List[ValueError]:
+    """
+    Returns a list of errors found in the authors list
+    """
+    all_errors = validate_authors_status(authors)
+
+    if running_network_validation:
+        orcids = list({author.ORCID for author in authors if author.ORCID is not None})
+
+        if len(orcids) == 0:
+            return all_errors
+
+        logger.debug("Checking ORCIDs: %s", orcids)
+        invalid_orcids = asyncio.run(validate_orcids(orcids))
```

Comment:
nit: we don't have to cast the set as a list. We can set the type in the method called as an `Iterable[str]` object. Also, one nice thing with if conditions in python is that they evaluate to `False` if an iterable variable is empty.


```
orcids = {author.ORCID for author in authors if not author.ORCID }
if not orcids:
     return all_errors
```